### PR TITLE
New recipe for Ninja build system

### PR DESCRIPTION
**Pull request by frol**
[Ninja](https://ninja-build.org/) is a small build system with a focus on speed. It differs from other build systems in two major respects: it is designed to have its input files generated by a higher-level build system, and it is designed to run builds as fast as possible.

### CODE REVIEW

**Comment by ocefpaf** [comment]
> Ninja is a small build system with a focus on speed. It differs from other build systems in two major respects: it is designed to have its input files generated by a higher-level build system, and it is designed to run builds as fast as possible.

I :heart: how you add recipes with a big summary of the package!

File: recipes/ninja/bld.bat [hunk 1]
```diff
@@ -0,0 +1,5 @@
+%PYTHON% configure.py --bootstrap
+if errorlevel 1 exit 1
+
+COPY ninja.exe %LIBRARY_BIN%\ninja.exe
```
Comment (ocefpaf) [inline recipes/ninja/bld.bat]: We are usually OK with that, but open an issue at the `feedstock` as soon as it is created mentioning a possible build from source.

Comment (ocefpaf) [inline recipes/ninja/bld.bat]: Oh no. Now that I see that is from source! Sorry.

**Comment by ocefpaf** [comment]
The Windows build is failing only on Python 3.5. Maybe our expert @gillins can help here :wink:

**Comment by frol** [comment]
This doesn't make any sense to me and since there is no run dependency, it is fine to build the package only on Python 2.7...

**Comment by msarahan** [comment]
This (a standalone EXE with no library dependencies) is a good use case for forcing the msvc_compiler.

```
build:
    msvc_compiler: 9.0

requirements:
    run:
        - vs2008_runtime
```

I don't know how to tell Appveyor to skip the other python versions.  Maybe with a skip section?

```
build:
    skip: True   # [win and py>27]
```
```

**Comment by frol** [comment]
@msarahan Hmm, I don't see anything like that in, for example, [cmake recipe](https://github.com/conda-forge/cmake-feedstock/blob/master/recipe/meta.yaml). Do I really have to add those options you mention in the first block?

I can certainly add the `skip` option, but is it the correct way of dealing with the weirdness on the latest Python version?

**Comment by msarahan** [comment]
For the CMake recipe, it's not there because it isn't building from source.  If you're building from source, you need to be more explicit about which compiler you use, so that you make sure that you have the appropriate runtime in place.

The skip option is weird, I agree.  If anyone has better ideas, I'm happy to discuss.  Maybe some kind of single architecture option, with the exact version depending on some system configuration.  It doesn't really matter which compiler these things are built with, so long as the build succeeds, and the correct runtime is listed as a dependency.

**Comment by frol** [comment]
I have updated the recipe taking your suggestions into account. Let's see how it goes green.

**Comment by frol** [comment]
All green!

**Comment by msarahan** [comment]
Great, I'm glad it worked!  Merging.

**Comment by ocefpaf** [comment]
:+1:

**Comment by jakirkham** [comment]
Neat. Thanks @frol.

Just a minor comment. Looks like their URL is now ( http://ninja-build.org ) at least according to their [repo]( https://github.com/ninja-build/ninja ). If we could please have that updated in the feedstock that would be great. Feel free to cross reference this with that change.

Thanks again @frol.

**Comment by frol** [comment]
@jakirkham Done. Thank you for pointing this out. I was moving these recipes from our Salford Systems' recipes repo, this is why the link was outdated.

**Comment by jakirkham** [comment]
Absolutely. We are happy to help you move everything over. I have done similar with a sizeable portions of my stack as well.

As I have been doing it, I have tried to use my PRs to staged-recipes as an opportunity to clean them up and update any metadata for the similar reasons. Namely, some of them I built awhile ago and I still use, but haven't needed to rebuild recently.

**Pull request merged**